File: testdata/bench/go/BenchmarkTest00030.go
package bench

import (
	"net/http"
	"os/exec"
)

// Command injection via multiple hops
func Handler00030(w http.ResponseWriter, r *http.Request) {
	raw := r.URL.Query().Get("prog")
	prog := raw
	arg := "--version"
	exec.Command(prog, arg).Run()
}

File: testdata/bench/kotlin/BenchmarkTest00030.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.http.*

// Vulnerable: XSS via query parameter in img src
suspend fun handler00030(call: ApplicationCall) {
    val url = call.request.queryParameters["avatar"]
    call.respondText("<img src='$url' alt='avatar'>", ContentType.Text.Html)
}
